e2e/cypress/tests/integration/enterprise/self-hosted/self_hosted_purchase_spec.ts

```diff
@@ -0,0 +1,382 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// e.g. not_cloud cloud because we always want to exclude running automatically
+// until we create the special self-hosted run setup
+// Stage: @prod
+// Group: @enterprise @not_cloud @cloud
```

Comment:
can we add @self-hosted tag?

I need your input here. All the jobs in cypress-automation currently running have [license set up ](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/blob/e2e-master-prod/.gitlab-ci.yml#L124)

to run this in CI, should we add condition to skip license?
I am aiming to run this and other self-hosted test

Reply:
I think we can add any tags we want, but it looks to me that nothing is done automatically with them unless we configure it in the cypress-ui-automation repo: https://git.internal.mattermost.com/qa/cypress-ui-automation/-/blob/e2e-master-prod/.gitlab-ci.yml#L44 .

I think `@self-hosted` is too broad of a tag. Maybe we could call it `@hosted_customer` instead to narrow the focus to parts of the app related to license purchase? One way I could see us using `@hosted_customer` is to add it on to the `--sortLast` group, even after `@license_removal` group. That way, it wouldn't interfere with other tests. Does that usage make sense to you. Are there other usages you are thinking of?